docs/rules/max-len.md

```diff
@@ -147,7 +147,7 @@ var longRegExpLiteral = /this is a really really really really really long regul
 Examples of **correct** code for this rule with the `{ "ignorePattern": true }` option:
```

Comment:
I think this line is also incorrect: `ignorePattern` is a string option, so `ignorePattern: true` doesn't make sense.

Reply:
maybe it make more sense to allow `string` and `regex` -- using `RegExp(string)` can be confusing, and sometimes different behavior. thoughts?

Reply:
I think there are multiple issues we're discussing in this PR -- maybe we should separate them:

* The current example in the documentation doesn't work, so I think we should fix it in this PR.
* It seems like the comment parsing for config comments might have confusing behavior with backslashes. (Maybe we should create a new issue for this.)
* It might be simpler for the rule to have a regex option rather than a string. (I think it's unlikely that this will work because we need to support JSON config files which don't support regex. However, feel free to create a new issue for it if you'd like.

Reply:
agree. updated the PR, so it will not block the merge.:)